PowerPointLabs/PowerPointLabs/ColorsLab/ColorsLabPaneWPF.xaml.cs

```diff
@@ -1257,8 +1297,62 @@ private void SaveDefaultColorPaneThemeColors()
 
         #endregion
 
+        #region Recent Colors
+
+        /// <summary>
+        /// Clear the panel to all white color.
+        /// </summary>
+        private void EmptyRecentColorsPanel()
+        {
+            try
+            {
+                if (this.GetCurrentSlide() != null)
+                {
+                    dataSource.RecentColorOne = Color.White;
+                    dataSource.RecentColorTwo = Color.White;
+                    dataSource.RecentColorThree = Color.White;
+                    dataSource.RecentColorFour = Color.White;
+                    dataSource.RecentColorFive = Color.White;
+                    dataSource.RecentColorSix = Color.White;
+                    dataSource.RecentColorSeven = Color.White;
+                    dataSource.RecentColorEight = Color.White;
+                    dataSource.RecentColorNine = Color.White;
+                    dataSource.RecentColorTen = Color.White;
+                    dataSource.RecentColorEleven = Color.White;
+                    dataSource.RecentColorTwelve = Color.White;
+                }
+            }
+            catch (Exception e)
+            {
+                ErrorDialogBox.ShowDialog("Recent Colors Panel Reset Failed", e.Message, e);
+            }
+        }
+
+        
+        /// <summary>
+        /// Save current recent colors panel to file.
+        /// </summary>
+        private void SaveColorPaneRecentColors()
+        {
+            dataSource.SaveRecentColorsInFile(_defaultRecentColorDirectory);
+        }
+
+        /// <summary>
+        /// Load recent panel from file, or clear the panel if unsuccessful.
+        /// </summary>
+        private void LoadRecentColorsPanel()
+        {
+            bool isSuccessful = dataSource.LoadRecentColorsFromFile(_defaultRecentColorDirectory);
+            if (!isSuccessful)
+            {
+                EmptyRecentColorsPanel();
+            }
+        }
+
+        #endregion
```

Comment:
Extra #endregion?

Reply:
Nope! Not extra. It closes one of the sub-regions.